parameter_tools/src/test/java/coppercore/parameter_tools/test/PathProviderTests.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    @BeforeEach
+    public void CleanTest() {
+        environmentHandler =
+                EnvironmentHandler.getEnvironmentHandler(pathProvider.resolvePath("config.json"));
+        testPathProvider = environmentHandler.getEnvironmentPathProvider();
+    }
+
+    @Test
+    public void getReadPathNormalBehaviorTest() {
+        Assertions.assertEquals(
+                pathProvider.getFullPath()
+                        + File.separator
+                        + "normal"
+                        + File.separator
+                        + "NormalFile.json",
+                testPathProvider.resolveReadPath("NormalFile.json"));
+    }
+
+    @Test
+    public void getReadPathDefualtingBehaviorTest() {
+        Assertions.assertEquals(
+                pathProvider.getFullPath()
+                        + File.separator
+                        + "defaults"
+                        + File.separator
+                        + "DefaultFile.json",
+                testPathProvider.resolveReadPath("DefaultFile.json"));
+    }
+
+    @Test
+    public void getReadPathMissingBehaviorTest() {
+        try {
+            testPathProvider.resolveReadPath("RandomNoneExistantFile.json");
+            Assertions.fail("Should throw error about file missing");
+        } catch (Exception e) {
+            Assertions.assertEquals(
+                    "RandomNoneExistantFile.json does not exist at "
+                            + pathProvider.getFullPath()
+                            + File.separator
+                            + "defaults"
+                            + File.separator
+                            + "RandomNoneExistantFile.json",
+                    e.getMessage());
+        }
+    }
+
+    @Test
+    public void getWritePathNormalBehaviorTest() {
+        Assertions.assertEquals(
+                pathProvider.getFullPath()
+                        + File.separator
+                        + "normal"
+                        + File.separator
+                        + "NormalFile.json",
+                testPathProvider.resolveWritePath("NormalFile.json"));
+    }
+
+    @Test
+    public void getWritePathDefualtingBehaviorTest() {
+        Assertions.assertEquals(
+                pathProvider.getFullPath()
+                        + File.separator
+                        + "defaults"
+                        + File.separator
+                        + "DefaultFile.json",
+                testPathProvider.resolveWritePath("DefaultFile.json"));
+    }
+
+    @Test
+    public void getWritePathDefualtingMissingTest() {
+        Assertions.assertEquals(
+                pathProvider.getFullPath()
+                        + File.separator
+                        + "defaults"
+                        + File.separator
+                        + "RandomNoneExistantFile.json",
+                testPathProvider.resolveWritePath("RandomNoneExistantFile.json"));
```

Comment:
RandomNonExistentFile (spelling)

this also appears in other parts of this file